core/include/gnuradio-4.0/Graph.hpp

```diff
@@ -431,39 +431,48 @@ class BlockWrapper : public BlockModel {
     }
 };
 
-class edge {
+template<typename T>
+struct PortIndexDefinition {
+    T           topLevel;
```

Comment:
If you could change `T` -> `std::size_t` and remove the unnecessary templatisation and constructor.

Reply:
PortIndexDefinition is used with T=std::size_t and T=std::string, as an argument to the connect() calls, so it's desirable from an API POV that the passed string/std::size_t parameters are implicitly converted to PortIndexDefinition. (thus the non-explicit ctor)

Reply:
OK.